src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -675,26 +564,43 @@ def sync_lock(self) -> Lock:
 
     def _initialize(self) -> None:
         """Initialize the SQL store."""
-        from zenml.zen_stores.metadata_store import MetadataStore
-
         logger.debug("Initializing SqlZenStore at %s", self.config.url)
 
-        metadata_config = self.config.get_metadata_config()
-        self._metadata_store = MetadataStore(config=metadata_config)
-
         url, connect_args, engine_args = self.config.get_sqlmodel_config()
         self._engine = create_engine(
             url=url, connect_args=connect_args, **engine_args
         )
+
+        # SQLite: As long as the parent directory exists, SQLAlchemy will
+        # automatically create the database.
+        local_path = self.config.get_local_path(url)
+        if local_path and not fileio.exists(local_path):
+            fileio.makedirs(os.path.dirname(local_path))
```

Comment:
```suggestion
        if (
            self.config.driver == SQLDatabaseDriver.SQLITE
            and self.config.database
            and not fileio.exists(self.config.database)
        ):
            fileio.makedirs(os.path.dirname(self.config.database))
```

And with this you can remove the `get_local_path` method from the configuration, as it's already overcrowded with all kinds of helper methods.

Reply:
Very nice, was looking for something like that but apparently missed it 👍